### PR TITLE
Add check_bbox argument management

### PR DESCRIPTION
**Pull request by EmmaRenauld**
Very small fix, remove bounding box valid checks.

### CODE REVIEW

**Review by AntoineTheb — APPROVED** [review]
Personnellement je mettrais ça _partout_ dans scilpy, mais c'est en effet une bonne idée d'ajouter ça au moins au scoring.

**Review by AntoineTheb — APPROVED** [review]
Personnellement je mettrais ça _partout_ dans scilpy, mais c'est en effet une bonne idée d'ajouter ça au moins au scoring.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Ok can you tell me a little bit about why this is such an issue ? Why we don't add a new arg to force bbox_valid_check to false ?

**Comment by EmmaRenauld** [comment]
I'm not sure in other scripts, but here we are scoring tractograms that are potentially bad. 

But if I try to think about general cases, I would say the most probable case is when vertical bundles/streamlines go too far in the brainstem under the pons, into the spinal cord, they can go out of the image quite easily. Other ideas, @AntoineTheb ?

**Comment by AntoineTheb** [comment]
Yeah that's usually the problem. The FiberCup's ground truth also goes out its volume.

**Comment by AntoineTheb** [comment]
Thinking about it, it could probably be a good idea to add an option to the `argparser` to disable bbox checking (and make it easily available to other scripts, like the `--verbose` or `--force` option for example). I'll add an issue to remind myself to do it @arnaudbore @EmmaRenauld .

**Comment by arnaudbore** [comment]
I agree, I don't like to automatically skip the bbox checking.

**Comment by EmmaRenauld** [comment]
I agree too.

**Comment by EmmaRenauld** [comment]
I made the change in all scripts using that check. Tested the whole pytest locally, it works.

**Comment by AntoineTheb** [comment]
Cool ! Do you have a tractogram to test it ?

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by EmmaRenauld** [comment]
```
scoring_data=$BrainData/databases/ismrm_2015/derivatives/scoring_data/output/version_2022
ref=$scoring_data/t1.nii.gz
config_file_segmentation=$scoring_data/config_file_segmentation.json
config_file_tractometry=$scoring_data/config_file_tractometry.json
submissions_dir=$BrainData/databases/ismrm_2015/derivatives/submissions/output/anonymized_submissions_converted_tck

out_dir_root=tmp/
mkdir $out_dir_root

tractogram=$submissions_dir/3_3.tck
team=3_3
out_dir=$out_dir_root/$team;

# ------ Segmentation -------
scil_score_tractogram.py -f -v $tractogram $config_file_segmentation $out_dir --no_empty \
          --gt_dir $scoring_data --reference $ref --json_prefix tmp_  # ----------------> Fails

scil_score_tractogram.py -f -v $tractogram $config_file_segmentation $out_dir --no_empty \
          --gt_dir $scoring_data --reference $ref --json_prefix tmp_ --no_bbox_check  # --------> works

# ------ Final scoring -------
scil_score_bundles.py -f -v $config_file_tractometry $out_dir \
            --gt_dir $scoring_data --reference $ref;  # ----------------> Fails

scil_score_bundles.py -f -v $config_file_tractometry $out_dir \
            --gt_dir $scoring_data --reference $ref --bbox_check False;  # ------------------> Works.
```

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Review by arnaudbore — COMMENTED** [review]
Looking at all these modifications I think we have to really discuss about this implication because a lot of save and load were using bbox_valid_check=False and also now there is no distinctions between load and save and I don't really know if it's a good thing or not.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: scripts/scil_extract_ushape.py [hunk 1]
```diff
@@ -70,8 +70,7 @@ def main():
         parser.error('Min-Max ufactor "{},{}" '.format(args.minU, args.maxU) +
                      'must be between -1 and 1.')
 
-    sft = load_tractogram_with_reference(
-        parser, args, args.in_tractogram)
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply scripts/scil_extract_ushape.py:73]: `bbox_check=False` was not there to start with. I just removed the unnecessary newline.

**Comment by EmmaRenauld** [comment]
Maybe I can remove this big commit for now, just so that scil_score_bundles can be usable in the next release, and we keep this bigger change for later?

**Comment by arnaudbore** [comment]
@EmmaRenauld yes go !

File: scripts/scil_extract_ushape.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AntoineTheb) [reply scripts/scil_extract_ushape.py:73]: I'm not sure what you mean with "no distinction between load and save" @arnaudbore

**Comment by EmmaRenauld** [comment]
> there is no distinctions between load and save and I don't really know if it's a good thing or not.

Same as Antoine. I'm not sure either what you mean. I did not change the way it already was. Even better, I removed the default "False" in many places to make it optional.

But to try and answer... In the case where you need to use it because you have a faulty tractogram, there is a very good change that if a streamline is misplaced in the input, it will also be in the output, no matter what the script does. So I would guess that it's ok in general.

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-641/6/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-641/7/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by EmmaRenauld** [comment]
Is this ready, now that the release is done? @arnaudbore

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by frheault** [comment]
@EmmaRenauld  Can you update this code with master and then check if everything runs normally.
The master is now up to 3.10 (but should run in 3.8/3.9), so you may have to create a new venv.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by frheault** [comment]
Out of curiosity, how did you pick which script to change immediately in this PR? Are they all script that runs even with invalid?

**Comment by EmmaRenauld** [comment]
Hmmm.. it was 6 months ago so I don't exactly remember, sorry. I probably searched the word bbox_valid_check in all files.

File: scripts/scil_decompose_connectivity.py [hunk 2]
```diff
@@ -274,8 +276,7 @@ def main():
 
     logging.info('*** Loading streamlines ***')
     time1 = time.time()
-    sft = load_tractogram_with_reference(parser, args, args.in_tractogram,
-                                         bbox_check=False)
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
     sft.remove_invalid_streamlines()
```
Comment (frheault) [inline scripts/scil_decompose_connectivity.py:280]: Here you should remove the invalid only if the check is off (if it is one, the check is done internally).
Skipping it if possible would save time.

File: scripts/scil_remove_invalid_streamlines.py [hunk 3]
```diff
@@ -63,8 +65,7 @@ def main():
     if args.threshold < 0:
         parser.error("Threshold must be positive.")
 
-    sft = load_tractogram_with_reference(parser, args, args.in_tractogram,
```
Comment (frheault) [inline scripts/scil_remove_invalid_streamlines.py:66]: Considering the nature of the script, I would leave as it was.
100% of the time a user will want the check OFF if they want to remove invalid.
So I would revert this one.

**Review by frheault — CHANGES_REQUESTED** [review]

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Pull request merged**